### PR TITLE
Make use of CryptoMetrics in DynamoDBCryptoKeyRepository

### PR DESCRIPTION
**Pull request by otbe**
+ ignore lombok generated code

### CODE REVIEW

File: cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java [hunk 1]
```diff
@@ -1,88 +1,97 @@
...
+        try {
+            val result = metrics.timed("createKeyInDynamoDbTable",
+                    () -> dynamoDB.updateItem(createRequest.toDynamoRequest())
+            );
+
+            val resultKey = Utils.extractCryptoKeyFromItem(algorithm, size, result.getAttributes());
+
+            if (!resultKey.isPresent()) {
+                // should never ever happen because that would indicate a broken DynamoDB API
+                // contract
```
Comment (uweschaefer) [inline cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java]: notify metrics about it?

File: cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java [hunk 2]
```diff
@@ -1,88 +1,97 @@
...
+        } catch (ConditionalCheckFailedException ignored) {
+            // this happens when the key was not found in the first step but someone created
+            // one in the meantime
+            // the updateItem call checks that the key for algorithm and size does not exist
+            // before updating/creating the item
+            // so we can safely (consistent) read from the table and get our key
+            val item = findKeyFor(subjectId, algorithm, size);
+
+            if (!item.isPresent()) {
+                throw new IllegalStateException("DynamoDB consistent read failed.");
```
Comment (uweschaefer) [inline cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java]: notify metrics about it?

File: cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java [hunk 3]
```diff
@@ -1,88 +1,97 @@
...
+    }
+
+    @Override
+    public CryptoKey getOrCreateKeyFor(@NonNull CryptoSubjectId subjectId, @NonNull CryptoAlgorithm algorithm,
+                                       @NonNull CryptoKeySize size) throws CryptoKeyNotFoundAfterCreatingException {
+        return findKeyFor(subjectId, algorithm, size).orElseGet(() -> createCryptoKey(subjectId, algorithm, size));
+    }
+
+    protected CryptoKey createCryptoKey(CryptoSubjectId subjectId, CryptoAlgorithm algorithm, CryptoKeySize size) {
+        metrics.notifyKeyCreation();
```
Comment (uweschaefer) [inline cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java]: should be done before returning?

File: cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java (hunk 2 above)
Comment (uweschaefer) [reply cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java]: or maybe not... maybe rename it to notifyKeyCreationAttempt then?

File: cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java (hunk 3 above)
Comment (otbe) [reply cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java]: Done

File: cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java (hunk 2 above)
Comment (uweschaefer) [reply cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java]: nobody talks to me... resolving :|

File: cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java (hunk 1 above)
Comment (uweschaefer) [reply cryptoshred-cloud-aws/src/main/java/eu/prismacapacity/cryptoshred/cloud/aws/DynamoDBCryptoKeyRepository.java]: nobody talks to me... resolving :|

**Pull request merged**